cpp/inc/bond/ext/grpc/detail/io_manager_tag.h

```diff
@@ -15,13 +15,18 @@ namespace bond { namespace ext { namespace gRPC { namespace detail {
     /// operation in its implementation of \ref invoke.
     struct io_manager_tag
     {
-        virtual ~io_manager_tag() { }
+        virtual ~io_manager_tag() = default;
 
         /// @brief Called when this instance is dequeued from a completion
         /// queue.
         ///
         /// @param ok whether or not the initial operation succeeded
         virtual void invoke(bool ok) = 0;
+
+        io_manager_tag* tag()
+        {
+            return this;
+        }
```

Comment:
Consider adding the reverse function to move the tag/untag logic all into `io_manager_tag`:

```cpp
/// Invokes a previously tagged completion queue operation.
///
/// @param tag A %tag value, as produced by the tag function, to resume.
/// @param ok whether or not the initial operation succeeded
static void invoke_tag(void* tag, bool ok)
{
    static_cast<io_manager_tag*>(tag)->invoke(ok);
}
```

Doing this also lets us make `tag()` return `void*`, making the tag opaquer.

Reply:
Good point. Carry on without.